cmd/list.go

```diff
@@ -0,0 +1,280 @@
+	}()
+
+	go func() {
+		// Get credentials
+		credentials, cErr = client.Credentials.Search(c, filterPathExp.String())
+		getEnvsServicesCreds.Done()
+	}()
+
+	getEnvsServicesCreds.Wait()
+
+	if cErr != nil {
+		return errs.NewErrorExitError("Could not retrieve credentials.", cErr)
+	}
+
+	if eErr != nil {
+		return errs.NewErrorExitError("Could not retrieve environments.", eErr)
+	}
+
+	if sErr != nil {
+		return errs.NewErrorExitError("Could not retrieve services.", sErr)
+	}
+
+	filteredEnvNames := []string{}
+	filteredServiceNames := []string{}
+
+	// Filter out the retrieved environments based on the
+	// search space provided in filterPathExp. If no flags
+	// were set, all environments will pass the following test.
+	for _, e := range environments {
+		if filterPathExp.Envs.Contains(e.Body.Name) {
+			filteredEnvNames = append(filteredEnvNames, e.Body.Name)
+		}
+	}
+
+	// Filter out the retrieved services based on the
+	// search space provided in filterPathExp. If no flags
+	// were set, all services will pass the following test.
+	for _, s := range services {
+		if filterPathExp.Services.Contains(s.Body.Name) {
+			filteredServiceNames = append(filteredServiceNames, s.Body.Name)
+		}
+	}
+
+	// Create credentialsTree
+	// This will be filled in the following section.
+	for _, eName := range filteredEnvNames {
+		credentialsTree[eName] = make(map[string]credentialSet)
+		for _, sName := range filteredServiceNames {
+			credentialsTree[eName][sName] = make(credentialSet)
+		}
+	}
+
+	// Check for each env and service in the filtered list, add
+	// any credentials along that path to that env/service branch
+	// of the credentialsTree
+	projectPath := "/" + orgName + "/" + projectName + "/"
+	for _, e := range filteredEnvNames {
+		for _, s := range filteredServiceNames {
+			builtPathExp, err := pathexp.Parse(projectPath + e + "/" + s + "/*/*")
+			for _, cred := range credentials {
+				if len(args) > 0 && isSecretNameInList((*cred.Body).GetName(), args) == false {
+					continue
+				}
+				credPathExp := (*cred.Body).GetPathExp()
+				if err != nil {
+					return errs.NewErrorExitError("Failed to parse: "+projectPath+e+"/"+s+"/*/*", err)
+				}
+				// If cred not contained in any builtPathExps, it is not
+				// within the search space specified by the flags.
+				if credPathExp.Contains(builtPathExp) == false {
+					continue
+				}
+				// "Add" is defined in 'credential_set.go'. This
+				// handles the case where a secret is redefined in
+				// overlapping spaces.
+				credentialsTree[e][s].Add(cred)
+			}
+		}
+	}
+
+	// Print credentialsTree
+	fmt.Println("")
+	w := tabwriter.NewWriter(os.Stdout, 5, 0, 2, ' ', 0)
+	fmt.Fprintf(w, "%s\n", projectPath)
+	for e := range credentialsTree {
+		fmt.Fprintf(w, "\t%s\t\n", e+"/")
+		for s := range credentialsTree[e] {
+			fmt.Fprintf(w, "\t\t%s\t\n", s+"/")
+			if len(credentialsTree[e][s]) == 0 && verbose {
+				fmt.Fprintf(w, "\t\t\t[empty]\n")
+			} else {
```

Comment:
You could simplify this using `continue`:

```
if len(credentialsTree[e][s]) == 0 {
  if verbose {
    fmt.Fprintf(w, "\t\t\t[empty]\n")
  }

  continue
}
```

Reply:
Yes, good thought, fixed